src/dmd/doc.d

```diff
@@ -598,7 +598,12 @@ extern (C++) void escapeStrayParenthesis(Loc loc, OutBuffer* buf, size_t start)
                 u++;
             }
             if (numdash >= 3)
-                inCode = !inCode;
+                inCode = inCode == c ? 0 : c;
```

Comment:
What is this change in logic doing?

Reply:
Not what I expected :)

Since there are actually two types of code sections, \`backtick-quoted spans\` and `---` fenced blocks, I meant to change the toggling logic to only toggle `inCode` off if it was the same type of code section as was started. However I did it wrong. I'll make a fix.